src/Microsoft.VisualStudio.ProjectSystem.Managed/MsBuildUtilities.cs

```diff
@@ -0,0 +1,172 @@
+            Requires.NotNull(evaluatedPropertyValue, nameof(evaluatedPropertyValue));
+            Requires.NotNullOrEmpty(propertyName, nameof(propertyName));
+
+            var property = GetOrAddProperty(project, propertyName);
+            StringBuilder newValue = new StringBuilder();
+            foreach (string value in GetPropertyValues(evaluatedPropertyValue, delimiter))
+            {
+                newValue.Append(value);
+                newValue.Append(delimiter);
+            }
+
+            newValue.Append(propertyValue);
+            property.Value = newValue.ToString();
+        }
+
+        /// <summary>
+        /// Renames a value inside a delimited property.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="evaluatedPropertyValue">Original evaluated value of the property.</param>
+        /// <param name="propertyName">Property name.</param>
+        /// <param name="propertyValue">Value to remove from the property.</param>
+        /// <param name="delimiter">Character used to delimit the property values.</param>
+        /// <remarks>
+        /// If the property is not present it will be added. This means that the evaluated property
+        /// value came from one of the project imports.
+        /// </remarks>
+        public static void RemovePropertyValue(ProjectRootElement project, string evaluatedPropertyValue, string propertyName, string propertyValue, char delimiter = ';')
+        {
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(evaluatedPropertyValue, nameof(evaluatedPropertyValue));
+            Requires.NotNullOrEmpty(propertyName, nameof(propertyName));
+
+            var property = GetOrAddProperty(project, propertyName);
+            StringBuilder newValue = new StringBuilder();
+            foreach (string value in GetPropertyValues(evaluatedPropertyValue, delimiter))
+            {
+                if (value != propertyValue)
+                {
+                    newValue.Append(value);
+                    newValue.Append(delimiter);
+                }
+            }
+
+            property.Value = newValue.ToString().TrimEnd(delimiter);
+        }
+
+        /// <summary>
+        /// Renames a value inside a delimited property.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="evaluatedPropertyValue">Original evaluated value of the property.</param>
+        /// <param name="propertyName">Property name.</param>
+        /// <param name="oldValue">Current property value.</param>
+        /// <param name="newValue">New property value.</param>
+        /// <param name="delimiter">Character used to delimit the property values.</param>
+        /// <remarks>
+        /// If the property is not present it will be added. This means that the evaluated property
```

Comment:
I think you're asking about the value not the property itself right? Three options - 

1) Do a no-op, can't rename something that doesn't exist in the first place.
2) Throw an error so that it can be handled upstream
3) Treat it as an add and append the value at the end

From a truly generic point of view the second option makes more sense. I went with the first though because this is (currently) used in the limited scope of something that's guaranteed to exist (because the config manager dialog is blocking so you can't edit the file at the same time as you're doing edits). I don't feel strongly about it though and I'd be happy to change it to either of the two other options if that sounds better.